### PR TITLE
Add Feature: get all articles

### PR DESCRIPTION
**Pull request by asifiqbal91**
## Add API support for getting articles list of a specific section.

**REF: [https://developer.zendesk.com/rest_api/docs/help_center/articles#list-articles](https://developer.zendesk.com/rest_api/docs/help_center/articles#list-articles)**

> List Articles endpoints:
> GET /api/v2/help_center/sections/{id}/articles.json
> GET /api/v2/help_center/{locale}/sections/{id}/articles.json

### CODE REVIEW

File: tests/Zendesk/API/UnitTests/HelpCenter/ArticlesTest.php [hunk 1]
```diff
@@ -12,13 +12,25 @@ class ArticlesTest extends BasicTest
...
         $articlesResource = $this->client->helpCenter->articles();
         $articlesResource->setLocale('en-US');
 
         $this->assertEndpointCalled(function () use ($articlesResource) {
             $articlesResource->findAll();
         }, 'help_center/en-US/articles.json');
-        
+
+        $this->assertEndpointCalled(function () use ($articlesResource) {
+            $articlesResource->find(1);
```
Comment (kcasas) [inline tests/Zendesk/API/UnitTests/HelpCenter/ArticlesTest.php]: User `$faker` here. Also can you separate this tests on another `method` that is specific for the `sections` part of the `Articles` object.

**Comment by asifiqbal91** [comment]
@kcasas $faker has added.

File: tests/Zendesk/API/UnitTests/HelpCenter/ArticlesTest.php [hunk 2]
```diff
@@ -12,28 +12,45 @@ class ArticlesTest extends BasicTest
      */
     public function testRoutesWithLocale()
     {
+        $faker = Factory::create();
+        $locale = $faker->locale;
+        $locale = $faker->locale;
```
Comment (samgavinio) [inline tests/Zendesk/API/UnitTests/HelpCenter/ArticlesTest.php]: nitpick, duplicate locale code here.

File: tests/Zendesk/API/UnitTests/HelpCenter/ArticlesTest.php [hunk 3]
```diff
@@ -12,28 +12,45 @@ class ArticlesTest extends BasicTest
...
+        $articleId = $faker->numberBetween(1);
+        $this->assertEndpointCalled(function () use ($articlesResource, $articleId) {
+            $articlesResource->find($articleId);
+        }, "help_center/{$locale}/articles/{$articleId}.json");
     }
 
     /**
      * Test if the route can be generated
      */
     public function testRouteWithLocale()
```
Comment (samgavinio) [inline tests/Zendesk/API/UnitTests/HelpCenter/ArticlesTest.php]: @jmramos02 I think this should be `testRouteWithoutLocale` perhaps you can rename this after this is merged?

**Pull request merged**